src/components/LinkToRecord/LinkToRecord.astro

```diff
@@ -12,6 +13,11 @@ export type Props = {
 const { record, openInNewTab, ...props } = Astro.props;
 const locale = getLocale() as SiteLocale;
 const href = getHref({ locale, record });
+const isType = (type: string) => record.__typename === type;
 ---
 
-<Link {href} {openInNewTab} {...props}><slot>{record.title}</slot></Link>
+{
+  (isType('FileRecord'))
+    ? <LinkToFile { record } {...props}><slot>{record.title}</slot></LinkToFile>
```

Comment:
This throws an error due to a number of things:
* spacing inside the braces results in typescript throwing an unhelpful type error
* `record` is of type `RecordRoute`, but `LinkToRecord` expects `FileRouteFragment`
* `FileRouteFragment` is a superset of `RecordRoute` adding `file` and `id`

Remove the function and Typescript will narrow it correctly

```suggestion
{
  (record.__typename === 'FileRecord')
    ? <LinkToFile {record} {...props}><slot>{title}</slot></LinkToFile>
```